scaleway/resource_mnq_sns_test.go

```diff
@@ -0,0 +1,142 @@
+package scaleway
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	accountV3 "github.com/scaleway/scaleway-sdk-go/api/account/v3"
+	mnq "github.com/scaleway/scaleway-sdk-go/api/mnq/v1beta1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func init() {
+	resource.AddTestSweepers("scaleway_mnq_sns", &resource.Sweeper{
+		Name: "scaleway_mnq_sns",
+		F:    testSweepMNQSNS,
+	})
+}
+
+func testSweepMNQSNS(_ string) error {
+	return sweepRegions((&mnq.SnsAPI{}).Regions(), func(scwClient *scw.Client, region scw.Region) error {
+		accountAPI := accountV3.NewProjectAPI(scwClient)
+		mnqAPI := mnq.NewSnsAPI(scwClient)
+
+		l.Debugf("sweeper: destroying the mnq sns in (%s)", region)
+
+		listProjects, err := accountAPI.ListProjects(&accountV3.ProjectAPIListProjectsRequest{}, scw.WithAllPages())
+		if err != nil {
+			return fmt.Errorf("failed to list projects: %w", err)
+		}
+		for _, project := range listProjects.Projects {
+			if !strings.HasPrefix(project.Name, "tf_tests") {
+				continue
+			}
+
+			_, err := mnqAPI.DeactivateSns(&mnq.SnsAPIDeactivateSnsRequest{
+				Region:    region,
+				ProjectID: project.ID,
+			})
+			if err != nil {
+				l.Debugf("sweeper: error (%s)", err)
+				return err
+			}
+		}
+
+		return nil
+	})
+}
+
+func TestAccScalewayMNQSNS_Basic(t *testing.T) {
```

Comment:
Can it be deactivated? Could you add a test step for that?

Reply:
Deactivation is the destroy step, it is checked by the CheckDestroy function